### ISSUE TITLE
App Crashes Immediately on Launch (MBP M1, macOS 11.5.2, Lunar 4.8.1)

### ISSUE THREAD

**Issue by jakematlick** [alin23/Lunar#405]
# Issue details
- Mac device where Lunar is installed (Macbook Pro 2019, iMac M1, Mac Mini M1, Hackintosh etc.):
Macbook Pro M1

- Monitor model (LG UltraFine 5K, DELL P2715Q etc.):
LG-32UD99-W (the crash happens with no external monitor attached as well)

- Monitor connection to the Mac device (HDMI-to-USB-C, USB-C-to-USB-C, miniDisplayPort-to-DisplayPort etc.):
USB-C-to-USB-C

- Using an USB Docking Station or Hub:
No

- Lunar mode used (check it in the top-right corner of the Lunar interface)
n/a (crash occurs on launch)

- Tried Lunar Diagnostics?
> 1. Click on **Open Lunar Diagnostics** in the Lunar menu
> 2. **Make sure to not click anywhere in the window**
> 3. Press keyboard keys when prompted by the text
> 4. After finishing the process click on the **Send Diagnostics** button.

no (crash occurs on launch)

# Issue description:
App crashes on launch. Attaching two crash logs... one when launching w/ external monitor connected and one w/out.

```
Application Specific Information:
BUG IN CLIENT OF LIBDISPATCH: trying to lock recursively

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libdispatch.dylib             	0x000000019550e074 _dispatch_once_wait.cold.1 + 28
1   libdispatch.dylib             	0x00000001954d9814 _dispatch_once_wait + 196
2   com.apple.AppKit              	0x0000000197fafa28 +[_NSScreenConfiguration latestZeroScreenHeight] + 120
3   com.apple.AppKit              	0x0000000197faf988 +[NSEvent mouseLocation] + 48
4   fyi.lunar.Lunar               	0x0000000105055da0 NSScreen.hasMouse.getter + 76
5   fyi.lunar.Lunar               	0x0000000105055b48 NSScreen.description.getter + 652
6   fyi.lunar.Lunar               	0x0000000105055880 @objc MPDisplayMode.description.getter + 32
7   com.apple.CoreFoundation      	0x00000001957719f4 -[NSArray descriptionWithLocale:indent:] + 552
8   com.apple.Foundation          	0x00000001964edf90 _NS_os_log_callback + 288
9   libsystem_trace.dylib         	0x0000000195425590 _os_log_fmt_flatten_NSCF + 64
10  libsystem_trace.dylib         	0x0000000195424ff4 _os_log_fmt_flatten_object + 316
11  libsystem_trace.dylib         	0x0000000195422f3c _os_log_impl_flatten_and_send + 1772
12  libsystem_trace.dylib         	0x0000000195422838 _os_log + 148
13  libsystem_trace.dylib         	0x000000019542279c _os_log_impl + 24
14  com.apple.AppKit              	0x0000000197f44778 _NSScreenConfigurationUpdateWithSharedInfo + 1788
15  com.apple.AppKit              	0x0000000197f429d0 ___NSScreenConfigurationEnsureInitialUpdateOccurred_block_invoke + 124
16  libdispatch.dylib             	0x00000001954d8ec0 _dispatch_client_callout + 20
17  libdispatch.dylib             	0x00000001954da758 _dispatch_once_callout + 32
18  com.apple.AppKit              	0x0000000197f42950 +[_NSScreenConfiguration prepare] + 56
19  com.apple.AppKit              	0x0000000197f4290c -[NSApplication(ScreenHandling) _registerForDisplayChangedNotifications] + 136
20  com.apple.AppKit              	0x0000000197f3a830 -[NSApplication init] + 1704
21  com.apple.AppKit              	0x0000000197f39f58 +[NSApplication sharedApplication] + 144
22  com.apple.AppKit              	0x0000000197f38600 NSApplicationMain + 432
23  fyi.lunar.Lunar               	0x0000000104fd6bec main + 12
24  libdyld.dylib                 	0x00000001956a6430 start + 4
```

[lunar-crash-logs.zip](https://github.com/alin23/Lunar/files/6989512/lunar-crash-logs.zip)

**Comment by jakematlick** [comment]
I just tried installing some older versions and 4.7.0 works and 4.7.1 and later crashes.

**Comment by alin23** [comment]
Thanks! I'll fix this for the 4.8.2 hotfix upgrade that will be released today

**Comment by alin23** [comment]
@jakematlick unfortunately I have no way of reproducing and testing the fix. 

Can you help me with testing this beta? https://static.lunar.fyi/Lunar-4.8.1-beta2.zip

**Comment by alin23** [comment]
I released v4.8.2 with this fix because some problems with hotkeys were more pressing.

Let me know if this is fixed now.

**Comment by jakematlick** [comment]
The problem is fixed in 4.8.2. Thank you!
